app/controllers/communicarts_controller.rb

```diff
@@ -21,10 +21,12 @@ def send_cart
   def approval_response
     cart = Cart.find(params[:cart_id]).decorate
     client_data = cart.proposal.client_data_legacy
-    approval = cart.approvals.find_by(user_id: user_id) 
+    approval = cart.approvals.find_by(user_id: user_id)
```

Comment:
Let's use `cart.approvals.approvable.find_by(user_id: user_id)` instead of the separate check below.